api/handler/v1beta1/job_spec.go

```diff
@@ -298,14 +299,45 @@ func (sv *JobSpecServiceServer) RefreshJobs(req *pb.RefreshJobsRequest, respStre
 		mu:     new(sync.Mutex),
 	})
 
-	if err := sv.jobSvc.Refresh(respStream.Context(), req.ProjectName, req.NamespaceNames, req.JobNames, observers); err != nil {
+	err := sv.jobSvc.Refresh(respStream.Context(), req.ProjectName, req.NamespaceNames, req.JobNames, observers)
+	if err != nil {
 		return status.Errorf(codes.Internal, "failed to refresh jobs: \n%s", err.Error())
 	}
 
 	sv.l.Info("finished job refresh", "time", time.Since(startTime))
 	return nil
 }
 
+func (sv *JobSpecServiceServer) GetDeployJobsStatus(ctx context.Context, req *pb.GetDeployJobsStatusRequest) (*pb.GetDeployJobsStatusResponse, error) {
+	deployID, err := uuid.FromString(req.DeployId)
+	if err != nil {
+		return nil, err
+	}
+
+	jobDeployment, err := sv.jobSvc.GetDeployment(ctx, models.DeploymentID(deployID))
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to get job deployment: \n%s", err.Error())
+	}
+
+	switch jobDeployment.Status {
+	case models.JobDeploymentStatusSucceed, models.JobDeploymentStatusFailed:
+		var deployJobFailures []*pb.DeployJobFailure
+		for _, failure := range jobDeployment.Details.Failures {
+			deployJobFailures = append(deployJobFailures, &pb.DeployJobFailure{JobName: failure.JobName, Message: failure.Message})
+		}
+
+		return &pb.GetDeployJobsStatusResponse{
```

Comment:
we can, I was not sure as it is kind of redundant data. for example, having failure_count and array of failures. although agree it will be straightforward metadata that can be used on the client side and for the user to process.

Reply:
updated, now we have success_count and failure_count